### ISSUE TITLE
Changed retention policy duration, queries return no data when querying for recent data

### ISSUE THREAD

**Issue by leafo** [influxdata/influxdb#7380]
### Bug report

__System info:__ InfluxDB 1.0.0, archlinux, built from AUR

__Steps to reproduce:__

I created a new retention policy when first creating my database, and set the duration to 2 weeks

```
> show retention policies on itchio
name		duration	shardGroupDuration	replicaN	default
autogen		0s		168h0m0s		1		false
two_weeks	336h0m0s	24h0m0s			1		true
```

After recording data for about 10 days, I updated the retention policy to keep data for much longer:

```
> alter retention policy two_weeks on itchio duration 365d 
> show retention policies on itchio
name		duration	shardGroupDuration	replicaN	default
autogen		0s		168h0m0s		1		false
two_weeks	8760h0m0s	168h0m0s		1     true
```

This was approximately 1 day ago. I noticed that if  queried for data within the past 2 days I would see drop off and no results at the same time I changed the retention policy. I thought that maybe it was rejecting writes, but writes appear to still be going through. If I query for a larger date range (over 3 days), then the data is available.

This query returns nothing:

```
> SELECT * FROM itchio.two_weeks."counts.ca.click.browse" WHERE time > now() - 1d  order by time desc limit 10
```

But this query returns the most recent data, still being written:

```
> SELECT * FROM itchio.two_weeks."counts.ca.click.browse" WHERE time > now() - 3d  order by time desc limit 10
name: counts.ca.click.browse
----------------------------
time			value
1475116788872134404	3
1475116758872784558	6
1475116728867548129	9
1475116698874245423	9
1475116668867557531	8
1475116638871759710	5
1475116608878057930	6
1475116578870915578	4
1475116548868590427	7
1475116518868062713	3
```

where right now is `1475116788872134404`

**Comment by leafo** [comment]
I changed the retention policy back to 2 weeks to see what would happen. The recent `- 1d` query returns data written since updating the policy again, but there's a huge gap in the time when it was set to 1 year. Selecting a larger time range still returns all the data.

**Comment by guanxudong** [comment]
+1 the same issue

**Comment by jwilder** [comment]
This is a dup of #5878 and will be fixed via #7370.

**Comment by leafo** [comment]
I saw that issue but after reading through it it appeared that the data was completely lost, where in this case the data is available but only when altering the query. Just highlighting that again in case they aren't the same issue. 

Will the fix in #7370 allow me to query the data normally? Or is it permanently messed up?

**Comment by jwilder** [comment]
When #7370 is merged, it should work correctly.  The data should not be not lost.  You may need to export the data for that shard with the missing time range and re-insert it to get it stored in the correct shard again.
